theme/icons.go

```diff
@@ -569,13 +569,16 @@ func (t *builtinTheme) Icon(n fyne.ThemeIconName) fyne.Resource {
 	return icons[n]
 }
 
-// ThemedResource is a resource wrapper that will return a version of the resource with the main color changed
-// for the currently selected theme.
+// ThemedResource is a resource wrapper used to represent a version of a resource.
+// It changes colors based on the active theme.
+// This is currently achieved using XML attributes.
+// This is designed for use with Icons of 24px x 24px.
```

Comment:
There is no limitation here - we use 24x24 SVG files, but that is not relevant here - also the "px" is misleading as the coordinate system is not pixels in SVG

Reply:
Are you sure? The following (two popular releases of the SVG spec) and a range of other sources are referring to pixels.

* https://dev.w3.org/SVG/profiles/1.1F2/publish/coords.html#InitialCoordinateSystem
* https://www.w3.org/TR/SVG/coords.html#InitialCoordinateSystem
* https://www.sarasoueidan.com/blog/svg-coordinate-systems/#:~:text=You%20specify%20the%20size%20of%20the%20viewport%20using%20the%20width%20and%20height%20attributes%20on%20the%20outermost%20%3Csvg%3E%20element.
* https://www.digitalocean.com/community/tutorials/svg-svg-viewbox

Reply:
Pretty sure, though it is subtle. From the spec you linked it does seem like I am wrong on first read:
> and one unit in the initial coordinate system equals one "pixel" (i.e., a px unit as defined in CSS2 ([CSS2], section 4.3.2)

However when you reflect a little more you see that a CSS pixel is not necessarily the same as a screen output pixel (which is what we mean in the very few instances that we mention pixel in Fyne). I.e. a raster pixel will be 1:1 with hardware, but CSS pixel will not:

i.e.: 
> Specifically, text, SVG and much of CSS will "just work" because of the automatic pixel scaling feature of the web

https://www.html5rocks.com/en/mobile/high-dpi/

Reply:
Well it will scale so no vector should ever be 1:1 device to screen pixel. Happy enough to remove `px` if it's confusing. For now I'm not sure here is the correct place to document. RN I may sink this PR; but leaving it to sit for a bit; with all these changes made in a local.

Reply:
> Well it will scale so no vector should ever be 1:1 device to screen pixel. Happy enough to remove `px` if it's confusing.

As we are in a vector toolkit the "pixel" word has specific meaning - i.e. its things that don't scale. A pixel here means an output pixel - and so the definition is not compatible with the "user scalable pixel" of SVG.

> For now I'm not sure here is the correct place to document. RN I may sink this PR; but leaving it to sit for a bit; with all these changes made in a local.

Tough to say - added docs in all places is good - some people come through API docs, others through the tutorials site so either can be valuable.

Reply:
> For now I'm not sure here is the correct place to document. RN I may sink this PR; but leaving it to sit for a bit; with all these changes made in a local.
> Tough to say - added docs in all places is good - some people come through API docs, others through the tutorials site so either can be valuable.

I would just like to add that API documentation is best added inside the repository. Some people, me included, prefer to use https://pkg.go.dev and if we generate other/extra API documentation somewhere else, it won't be shown there.